server/src/main/java/org/apache/uniffle/server/netty/ShuffleServerNettyHandler.java

```diff
@@ -442,8 +442,9 @@ public void handleGetLocalShuffleData(TransportClient client, GetLocalShuffleDat
                     length);
         long readTime = System.currentTimeMillis() - start;
         ShuffleServerMetrics.counterTotalReadTime.inc(readTime);
-        ShuffleServerMetrics.counterTotalReadDataSize.inc(sdr.getData().length);
-        ShuffleServerMetrics.counterTotalReadLocalDataFileSize.inc(sdr.getData().length);
+        int dataLength = sdr.getDataBuffer().remaining();
```

Comment:
Why not exposing the `bufferDataLen` method for being invoked?

Reply:
It's ok. We can create a refactor issue for this point.

Reply:
https://github.com/apache/incubator-uniffle/issues/1164 .Created.